services/cadexsoft/app/main.py

```
@@ -0,0 +1,196 @@
+    """Accept either JSON body or multipart upload, return raw bytes + metadata."""
+    if upload is not None:
+        data = await upload.read()
+        if len(data) > MAX_UPLOAD_BYTES:
+            raise HTTPException(status_code=413, detail="file exceeds max upload size")
+        opts: dict = {}
+        if options_form:
+            import json
+            try:
+                opts = json.loads(options_form)
+            except json.JSONDecodeError:
+                raise HTTPException(status_code=400, detail="invalid JSON in options field")
```

Ruff (0.15.10)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)